### PR TITLE
Created test environment and test cases

### PR DESCRIPTION
**Pull request by clemens29**
- added test enviroment + readme
- created test cases 03-05 #153 
- added testconcept (still in work)

### CODE REVIEW

File: src/backend/app/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+````
+
+Navigate to the `src` directory
+```bash
+docker exec -it django_backend bash
+````
+
+Run all tests
+```bash
+python manage.py test
```
Comment (henrikalbrecht) [inline src/backend/app/tests/README.md:15]: When I run the tests 1 fails.

```bash
root@38a11f3d6b3c:/app# python manage.py test
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
FE
======================================================================
ERROR: test_new_valid_vm (app.tests.testModels.VMTestCase)
@brief Test case for creating a valid VM instance.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(30)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/app/tests/testModels.py", line 35, in test_new_valid_vm
    new_vm.save()
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 739, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 776, in save_base
    updated = self._save_table(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 881, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 919, in _do_insert
    return manager._insert(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 1270, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1416, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.DataError: value too long for type character varying(30)


======================================================================
FAIL: test_new_invalid_vm (app.tests.testModels.VMTestCase)
@brief Test case for creating an invalid VM instance.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/app/tests/testModels.py", line 66, in test_new_invalid_vm
    new_vm.save()
AssertionError: Exception not raised

----------------------------------------------------------------------
Ran 2 tests in 0.039s

FAILED (failures=1, errors=1)
Destroying test database for alias 'default'...
```

**Comment by henrikalbrecht** [comment]
Is it possible to have all the test files as a sibling directory of source, it is not a good practice to have tests inside the `src` directory.

File: src/backend/app/tests/README.md (hunk 1 above)
Comment (clemens29) [reply src/backend/app/tests/README.md:15]: The test fails since the branch is pretty old... should I just merge the main into it?

**Comment by clemens29** [comment]
> Is it possible to have all the test files as a sibling directory of source, it is not a good practice to have tests inside the `src` directory.

DONE

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# How to use the test envirement
```
Comment (henrikalbrecht) [inline test/README.md:1]: small typo: "environment"

**Pull request merged**